integration/mediation-tests/tests-ui/src/test/java/org/wso2/carbon/esb/ui/test/templates/TemplateMediatorUITestCase.java

```diff
@@ -0,0 +1,141 @@
+*  in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*  http://www.apache.org/licenses/LICENSE-2.0
+*
+*  Unless required by applicable law or agreed to in writing,
+*  software distributed under the License is distributed on an
+*  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+*  KIND, either express or implied.  See the License for the
+*  specific language governing permissions and limitations
+*  under the License.
+*/
+package org.wso2.carbon.esb.ui.test.templates;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.ui.Select;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.automation.extensions.selenium.BrowserManager;
+import org.wso2.esb.integration.common.clients.registry.ResourceAdminServiceClient;
+import org.wso2.esb.integration.common.ui.page.LoginPage;
+import org.wso2.esb.integration.common.utils.ESBIntegrationUITest;
+
+/**
+ * Test the mediation template service which was used by the management console template creation wizard
+ */
+public class TemplateMediatorUITestCase extends ESBIntegrationUITest {
+    private WebDriver driver;
+    private ResourceAdminServiceClient resourceAdminServiceClient;
+
+    /**
+     * Initialize drive and set the console URL
+     *
+     * @throws Exception
+     */
+    @BeforeClass(alwaysRun = true)
+    public void setUp() throws Exception {
+        super.init();
+        driver = BrowserManager.getWebDriver();
+        driver.get(getLoginURL());
+        resourceAdminServiceClient = new ResourceAdminServiceClient(contextUrls.getBackEndUrl(), getSessionCookie());
+    }
+
+    /**
+     * Test for template create, edit and delete functions
+     *
+     * @throws Exception
+     */
+    @Test(groups = "wso2.esb",
+          description = "Verify that endpoints page renders when an invalid dynamic endpoint is present.")
+    public void testTemplatesListing() throws Exception {
+        LoginPage loginPage = new LoginPage(driver);
+        loginPage.loginAs(userInfo.getUserName(), userInfo.getPassword());
+
+        //go to template listing
+        driver.findElement(By.linkText("Templates")).click();
+
+        //create a new template
+        By newTemplateButoon = By.linkText("Add Sequence Template");
+        Assert.assertTrue(isElementPresent(newTemplateButoon),
+                          "Unable to find the button to create a new sequence template");
+        driver.findElement(newTemplateButoon).click();
+        driver.findElement(By.id("sequence.name")).clear();
+        driver.findElement(By.id("sequence.name")).sendKeys("sampleTemplate");
+
+        //create template configuration
+        driver.findElement(By.linkText("Add Child")).click();
+        driver.findElement(By.linkText("Log")).click();
```

Comment:
The element, "Log" is not visible here. We need to first select the element "Core"